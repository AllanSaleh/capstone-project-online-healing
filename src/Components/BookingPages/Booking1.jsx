import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import QuestionComponent from './QuestionComponent';

export default function Booking1() {
  window.scrollTo(0, 0);
  const history = useHistory();
  const { t } = useTranslation();

  const questionData = {
    question: t('Booking.Q1'),
    choices: [
      {
        content: t('Booking.Choice11'),
      },
      {
        content: t('Booking.Choice12'),
      },
    ],
  };

  const [answer, setAnswer] = useState('');
  const handleSetAnswer = (childAnswer) => {
    setAnswer(childAnswer);
    console.log(answer);
  };
  console.log(answer);
  const NextPage = () => {
    // Put firestore code here!
    history.push('/Booking2');
  };

  return (
    <div className="lg:h-firstsection px-sides pt-navbar">
      <h1 className="text-3xl lg:text-title uppercase my-8">{t('Booking.Title1')}</h1>
      <h3 className="text-md lg:text-subtitle opacity-50 text-justify">{t('Booking.Subtitle1')}</h3>

      <div className="flex flex-col justify-between w-full md:max-w-md lg:max-w-2xl my-16 mx-auto px-8 py-4 shadow-md">
        <QuestionComponent
          question={questionData.question}
          choices={questionData.choices}
          setParentAnswer={handleSetAnswer}
        />

        <div className="flex justify-end mt-16">
          <button
            onClick={() => NextPage()}
            type="button"
            className=" uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
          >{t('Booking.Next')}</button>
        </div>
      </div>
    </div>
  );
}
