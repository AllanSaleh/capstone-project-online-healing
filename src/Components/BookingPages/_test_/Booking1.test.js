import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Booking1 from '../Booking1';

it('Booking1 renders correctly', () => {
  const tree = renderer.create(<Booking1 />).toJSON();
  expect(tree).toMatchSnapshot();
});
