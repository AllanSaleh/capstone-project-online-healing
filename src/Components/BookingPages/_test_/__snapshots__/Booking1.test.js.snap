// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking1 renders correctly 1`] = `
<div
  className="lg:h-firstsection px-sides pt-navbar"
>
  <h1
    className="text-3xl lg:text-title uppercase my-8"
  >
    Booking.Title1
  </h1>
  <h3
    className="text-md lg:text-subtitle opacity-50 text-justify"
  >
    Booking.Subtitle1
  </h3>
  <div
    className="flex flex-col justify-between w-full md:max-w-md lg:max-w-2xl my-16 mx-auto px-8 py-4 shadow-md"
  >
    <ul>
      <h1
        className="text-lg lg:text-subtitle"
      >
        Booking.Q1
      </h1>
      <button
        className="w-full my-4 p-4 lg:p-6 text-left text-md lg:text-paragraph hover:cursor-pointer bg-white border border-black border-opacity-20 rounded-lg"
        onClick={[Function]}
        type="button"
      >
        Booking.Choice11
      </button>
      <button
        className="w-full my-4 p-4 lg:p-6 text-left text-md lg:text-paragraph hover:cursor-pointer bg-white border border-black border-opacity-20 rounded-lg"
        onClick={[Function]}
        type="button"
      >
        Booking.Choice12
      </button>
    </ul>
    <div
      className="flex justify-end mt-16"
    >
      <button
        className=" uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
        onClick={[Function]}
        type="button"
      >
        Booking.Next
      </button>
    </div>
  </div>
</div>
`;
